### PR TITLE
Resolvers for item, collection and community home page

### LINKED ISSUE #284: Use resolvers to fetch data
Resolvers are a concept in angular that allows you to fetch the data needed for a route before it loads. They were added to angular while dspace 7 was already in development, but we haven’t started using them. Refactoring to use resolvers will improve separation of concerns, code reusability and will likely speed up initial page loads.

### PR DESCRIPTION
**Pull request by LotteHofstede**
This PR closes #284 
I added resolvers for the item, collection and community home page to fetch the object's data before loading the rest of the page.

This PR also contains a fix for the recent items shown on the collection page. They are now requested using the `search` endpoint instead of the `items` endpoint.

### CODE REVIEW

File: config/environment.default.js [hunk 1]
```diff
@@ -8,13 +8,13 @@ module.exports = {
     nameSpace: '/'
   },
   // The REST API server settings.
-  rest: {
+    rest: {
```
Comment (tdonohue) [inline config/environment.default.js]: This indentation is slightly odd. Looks to be accidental?  (There's another such indentation in this same config file)

File: src/app/+collection-page/collection-page.component.html [hunk 2]
```diff
@@ -35,17 +35,17 @@
         </ds-comcol-page-content>
       </div>
     </div>
-    <ds-error *ngIf="collectionRD?.hasFailed" message="{{'error.collection' | translate}}"></ds-error>
+    <ds-error *ngIf="collectionRD?.hasFailed"0 message="{{'error.collection' | translate}}"></ds-error>
```
Comment (tdonohue) [inline src/app/+collection-page/collection-page.component.html]: There's a random `0` appended here?

File: src/app/+collection-page/collection-page.component.html [hunk 3]
```diff
@@ -35,17 +35,17 @@
...
-    <ng-container *ngVar="(itemRDObs | async) as itemRD">
+    <ng-container *ngVar="(itemRD$ | async) as itemRD">
       <div *ngIf="itemRD?.hasSucceeded" @fadeIn>
         <h2>{{'collection.page.browse.recent.head' | translate}}</h2>
         <ds-viewable-collection
           [config]="paginationConfig"
           [sortConfig]="sortConfig"
           [objects]="itemRD"
-          [hideGear]="false"
+          [hideGear]="true"
```
Comment (tdonohue) [inline src/app/+collection-page/collection-page.component.html]: Is there a reason to switch this from `false` to `true`?  It seems unrelated to this PR, but maybe I'm missing something?

File: src/app/+collection-page/collection-page.resolver.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, Resolve, RouterStateSnapshot } from '@angular/router';
+import { Collection } from '../core/shared/collection.model';
+import { Observable } from 'rxjs/Observable';
+import { CollectionDataService } from '../core/data/collection-data.service';
+import { RemoteData } from '../core/data/remote-data';
+import { getSucceededRemoteData } from '../core/shared/operators';
+
+@Injectable()
+export class CollectionPageResolver implements Resolve<RemoteData<Collection>> {
```
Comment (tdonohue) [inline src/app/+collection-page/collection-page.resolver.ts]: Please add TypeDocs to this new class

File: src/app/+community-page/community-page.resolver.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, Resolve, RouterStateSnapshot } from '@angular/router';
+import { Observable } from 'rxjs/Observable';
+import { RemoteData } from '../core/data/remote-data';
+import { getSucceededRemoteData } from '../core/shared/operators';
+import { Community } from '../core/shared/community.model';
+import { CommunityDataService } from '../core/data/community-data.service';
+
+@Injectable()
+export class CommunityPageResolver implements Resolve<RemoteData<Community>> {
```
Comment (tdonohue) [inline src/app/+community-page/community-page.resolver.ts]: Please add TypeDocs to this new class

File: src/app/+item-page/item-page.resolver.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, Resolve, RouterStateSnapshot } from '@angular/router';
+import { Observable } from 'rxjs/Observable';
+import { RemoteData } from '../core/data/remote-data';
+import { getSucceededRemoteData } from '../core/shared/operators';
+import { ItemDataService } from '../core/data/item-data.service';
+import { Item } from '../core/shared/item.model';
+
+@Injectable()
+export class ItemPageResolver implements Resolve<RemoteData<Item>> {
```
Comment (tdonohue) [inline src/app/+item-page/item-page.resolver.ts]: Please add TypeDocs to this new class

**Comment by LotteHofstede** [comment]
Thanks for your feedback, Tim
I added the missing TypeDoc for the resolvers and removed the typos.
The reason I decided to hide the gear is because I was trying to make sense of what this Recent Submissions section on a collection page (for which I changed the implementation) should actually display.
It does not make sense to give users the possibility to change the sorting configuration: Recent Submissions implies it's sorted by submission date. This feature would make sense on a browse by collection page.

**Pull request merged**